src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseHybridFVM.cpp

```diff
@@ -0,0 +1,899 @@
+                                                                           faceToNodes,
+                                                                           elemCenter,
+                                                                           elemVolume,
+                                                                           elemPerm,
+                                                                           elemGravCoef,
+                                                                           elemToFaces,
+                                                                           transMultiplier,
+                                                                           lengthTolerance,
+                                                                           mimFaceGravCoefNumerator.toView(),
+                                                                           mimFaceGravCoefDenominator.toView(),
+                                                                           mimFaceGravCoef );
+
+  } );
+
+}
+
+void CompositionalMultiphaseHybridFVM::ImplicitStepSetup( real64 const & time_n,
+                                                          real64 const & dt,
+                                                          DomainPartition & domain )
+{
+  GEOSX_MARK_FUNCTION;
+
+  // setup the elem-centered fields
+  CompositionalMultiphaseBase::ImplicitStepSetup( time_n, dt, domain );
+
+  // setup the face fields
+  MeshLevel & meshLevel = *domain.getMeshBodies()->GetGroup< MeshBody >( 0 )->getMeshLevel( 0 );
+  FaceManager & faceManager = *meshLevel.getFaceManager();
+
+  // get the accumulated pressure updates
+  arrayView1d< real64 > const & dFacePres =
+    faceManager.getReference< array1d< real64 > >( viewKeyStruct::deltaFacePressureString );
+
+  // zero out the face pressures
+  dFacePres.setValues< parallelDevicePolicy<> >( 0.0 );
+}
+
+
+void CompositionalMultiphaseHybridFVM::ImplicitStepComplete( real64 const & time_n,
+                                                             real64 const & dt,
+                                                             DomainPartition & domain )
+{
+  GEOSX_MARK_FUNCTION;
+
+  // increment the elem-centered fields
+  CompositionalMultiphaseBase::ImplicitStepComplete( time_n, dt, domain );
+
+  // increment the face fields
+  MeshLevel & meshLevel = *domain.getMeshBodies()->GetGroup< MeshBody >( 0 )->getMeshLevel( 0 );
+  FaceManager & faceManager = *meshLevel.getFaceManager();
+
+  // get the face-based pressure
+  arrayView1d< real64 > const facePres =
+    faceManager.getReference< array1d< real64 > >( viewKeyStruct::facePressureString );
+  arrayView1d< real64 > const dFacePres =
+    faceManager.getReference< array1d< real64 > >( viewKeyStruct::deltaFacePressureString );
+
+  forAll< parallelDevicePolicy<> >( faceManager.size(), [=] GEOSX_HOST_DEVICE ( localIndex const iface )
+  {
+    facePres[iface] += dFacePres[iface];
+    dFacePres[iface] = 0.0;
```

Comment:
We typically don't zero out solution variables on `Complete` as it is done in `Setup`... unless you have a reason to do it here.

Reply:
I agree. I removed this line from the `Complete` function.